routing/router.go

```diff
@@ -228,6 +228,9 @@ type ChannelPolicy struct {
 	MinHTLC *lnwire.MilliSatoshi
 }
 
+// RouteTransformFunc defines a function type for transforming a route.
+type RouteTransformFunc func(route *route.Route) *route.Route
```

Comment:
do we definitely want the flexibility to transform the entire route? It's a bit of a red flag to me cause we are depending on this module to construct a route but then we also as a caller are able to just replace that entire route... so can we not restrict the options a bit? Like `WithPeeledFirstNode` and things like that?